File: backend/internal/handlers/people.go
package handlers

import (
	"net/http"

	"github.com/gin-gonic/gin"

	"github.com/henrysh85/operation-d-g-g-op/backend/internal/models"
	"github.com/henrysh85/operation-d-g-g-op/backend/internal/repo"
)

type PeopleHandler struct{ Repo *repo.PeopleRepo }

func NewPeopleHandler(r *repo.PeopleRepo) *PeopleHandler { return &PeopleHandler{Repo: r} }

func (h *PeopleHandler) List(c *gin.Context) {
	out, err := h.Repo.List(c.Request.Context(), repo.PeopleFilter{
		Dept:   c.Query("dept"),
		Status: c.Query("status"),
		Search: c.Query("q"),
		Limit:  qInt(c, "limit", 200),
		Offset: qInt(c, "offset", 0),
	})
	if err != nil {
		HandleErr(c, err)
		return
	}
	c.JSON(http.StatusOK, gin.H{"data": out})
}

func (h *PeopleHandler) Get(c *gin.Context) {
	p, err := h.Repo.Get(c.Request.Context(), c.Param("id"))
	if err != nil {
		HandleErr(c, err)
		return
	}
	c.JSON(http.StatusOK, p)
}

func (h *PeopleHandler) Create(c *gin.Context) {
	var p models.Person
	if err := c.ShouldBindJSON(&p); err != nil {
		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
		return
	}
	if err := h.Repo.Create(c.Request.Context(), &p); err != nil {
		HandleErr(c, err)
		return
	}
	c.JSON(http.StatusCreated, p)
}

func (h *PeopleHandler) Patch(c *gin.Context) {
	var body map[string]any
	if err := c.ShouldBindJSON(&body); err != nil {
		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
		return
	}
	allowed := map[string]bool{
		"name": true, "email": true, "dept": true, "title": true, "reports_to": true,
		"status": true, "location": true, "country_code": true, "salary": true,
		"currency": true, "holiday_quota": true, "photo_key": true, "start_date": true,
		"metadata": true,
	}
	fields := map[string]any{}
	for k, v := range body {
		if allowed[k] {
			fields[k] = v
		}
	}
	if err := h.Repo.Patch(c.Request.Context(), c.Param("id"), fields); err != nil {
		HandleErr(c, err)
		return
	}
	c.JSON(http.StatusOK, gin.H{"ok": true})
}

func (h *PeopleHandler) Delete(c *gin.Context) {
	if err := h.Repo.Delete(c.Request.Context(), c.Param("id")); err != nil {
		HandleErr(c, err)
		return
	}
	c.Status(http.StatusNoContent)
}

File: frontend/src/api/people.ts
import { http, list as listHelper } from './client';
import type { Person } from '@/types';

interface BackendPerson {
  id: string;
  name: string;
  dept?: string;
  title?: string;
  location?: string;
  email?: string;
  reports_to?: string | null;
  photo_key?: string | null;
  start_date?: string | null;
  status?: string;
}

function adapt(p: BackendPerson): Person {
  return {
    id: p.id,
    fullName: p.name,
    role: p.title ?? '',
    team: p.dept ?? '',
    region: p.location ?? '',
    email: p.email ?? '',
    reportsTo: p.reports_to ?? null,
    photoKey: p.photo_key ?? null,
    joinedAt: p.start_date ?? null,
    status: p.status,
  } as unknown as Person;
}

export const people = {
  async list(params?: Record<string, unknown>): Promise<Person[]> {
    const rows = await listHelper<BackendPerson>('/people', params);
    return rows.map(adapt);
  },
  async get(id: string): Promise<Person> {
    const { data } = await http.get<BackendPerson>(`/people/${id}`);
    return adapt(data);
  },
  async create(payload: Partial<Person>): Promise<Person> {
    const { data } = await http.post<BackendPerson>('/people', payload);
    return adapt(data);
  },
  async update(id: string, payload: Partial<Person>): Promise<Person> {
    const { data } = await http.patch<BackendPerson>(`/people/${id}`, payload);
    return adapt(data);
  },
  async remove(id: string): Promise<void> {
    await http.delete(`/people/${id}`);
  },
};
